riscv/extension.h

```diff
@@ -35,4 +51,27 @@ void register_extension(const char* name, std::function<extension_t*()> f);
     public: register_##name() { register_extension(#name, constructor); } \
   }; static register_##name dummy_##name;
 
+class xs_gatherer_t
```

Comment:
Commit d5a80cf4c26fde5d01233156c5643b286c680043 is far too big and needs to be broken up into smaller steps. Such as:

1. Introduce and instantiate new (empty) class `xs_gatherer_t`
2. Register all extensions in class `xs_gatherer_t` (only adding it to the local `custom_extensions` variable; leave old `register_extension()` in place until obsoleted)
3. Move `get_extension()` to class `xs_gatherer_t`
4. Move `any_custom_extensions()` to class `xs_gatherer_t`
5. Add `set_debug()` to class `xs_gatherer_t`
6. Move remaining body of old `register_extension()` to new `xs_gatherer_t::register_extension()`
7. Delete old `register_extension()` and `processor_t::custom_extensions` now that they're no longer needed
8. Add extension state tracking to `xs_gatherer_t`